### ISSUE TITLE
用别的数据集会出现过拟合的现象

### ISSUE THREAD

**Issue by yzw-yzw** [FlagAlpha/Llama2-Chinese#156]
用llama-chinese提供的train_sft.csv数据集，过多（九千多条）过少（几百条）会出现过拟合的现象，两千多条不会。用别的数据集不论是通用知识数据集还是专业领域数据集（如交通和医疗）会出现过拟合的现象，eval_loss都是先降到1.0左右，然后上升。train_loss一直在下降，eval_accuracy一开始就在0.92左右，后续在这范围波动，没有特别明显的增长下降。训练出来的模型回答效果也不好，通用知识会张冠李戴，专业知识会答得风马牛不相及
其中测试集取训练集的前5%~8%，具体参数如下
!deepspeed  --include localhost:0 --master_port 29510 finetune_clm_lora.py \
    --model_name_or_path /root/autodl-tmp/Llama-2-7b-hf \
    --train_files /root/autodl-tmp/Llama2-Chinese/data/pdf/book.csv \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --do_train \
    --do_eval \
    --use_fast_tokenizer false \
    --output_dir /root/autodl-tmp/Llama2-Chinese/sft_7b_chat \
    --evaluation_strategy  steps \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps 16 \
    --num_train_epochs 10 \
    --warmup_steps 400 \
    --load_in_bits 4 \
    --lora_r 8 \
    --lora_alpha 32 \
    --target_modules q_proj,k_proj,v_proj,o_proj,down_proj,gate_proj,up_proj \
    --logging_dir /root/autodl-tmp/Llama2-Chinese/sft_7b_chat/logs \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --preprocessing_num_workers 10 \
    --save_steps 500 \
    --eval_steps 50 \
    --save_total_limit 3 \
    --seed 42 \
    --disable_tqdm false \
    --ddp_find_unused_parameters false \
    --block_size 4096 \
    --report_to tensorboard \
    --overwrite_output_dir \
    --deepspeed ds_config_zero2.json \
    --ignore_data_skip true \
    --bf16 \
    --gradient_checkpointing \
    --bf16_full_eval \
    --ddp_timeout 18000000

**Comment by ZHangZHengEric** [comment]
过拟合是很正常的现象，在小数据集上，transformers 的拟合能力还是很强的。我们提供的训练集，已经在模型里训练过，只不过是给大家的展示。
训练效果还涉及多方面的因素，如果有任何问题，可以进入微信群进行讨论
